### PR TITLE
[poco] Update and improve patches.

### LINKED ISSUE #27975: [poco] build failure with mariadb
**Host Environment**

- OS: [Windows]
- Compiler: VS2019
-vcpkg: master branch 

Installing 1/3 poco:x86-windows...
Building poco[core,mariadb,netssl]:x86-windows...
-- Using cached pocoproject-poco-1211613642269b7d53bea58b02de7fcd25ece3b9.tar.gz.
-- Cleaning sources at C:/vcpkg/buildtrees/poco/src/cd25ece3b9-a8fdd8b2f5.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/vcpkg/downloads/pocoproject-poco-1211613642269b7d53bea58b02de7fcd25ece3b9.tar.gz
-- Applying patch 0001-static-pcre.patch
-- Applying patch 0002-arm64-pcre.patch
-- Applying patch 0003-fix-dependency.patch
-- Applying patch 0004-fix-feature-sqlite3.patch
-- Applying patch 0005-fix-error-c3861.patch
-- Applying patch 0006-fix-install-data-mysql.patch
-- Applying patch 0007-find-pcre2.patch
-- Using source at C:/vcpkg/buildtrees/poco/src/cd25ece3b9-a8fdd8b2f5.clean
-- Found external ninja('1.10.2').
-- Configuring x86-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:96 (message):
    Command failed: "C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" -v
    Working Directory: C:/vcpkg/buildtrees/poco/x86-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      C:\vcpkg\buildtrees\poco\config-x86-windows-CMakeCache.txt.log
      C:\vcpkg\buildtrees\poco\config-x86-windows-CMakeCache.txt.log
      C:\vcpkg\buildtrees\poco\config-x86-windows-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:247 (vcpkg_execute_required_process)
  ports/poco/portfile.cmake:68 (vcpkg_cmake_configure)
  scripts/ports.cmake:147 (include)
[config-x86-windows-out.log](https://github.com/microsoft/vcpkg/files/10074091/config-x86-windows-out.log)

### PR DESCRIPTION
**Pull request by FrankXie05**
**Describe the pull request**

- #### What does your PR fix?
  Fixes https://github.com/microsoft/vcpkg/issues/27975
  Note:  Correct patch `0003-fix-dependency.patch`  https://github.com/microsoft/vcpkg/pull/27288 and merge patches.

### CODE REVIEW

**Review by MonicaLiu0311 — APPROVED** [review]

File: ports/poco/0007-find-pcre2.patch [hunk 1]
```diff
@@ -1,8 +1,8 @@
 diff --git a/CMakeLists.txt b/CMakeLists.txt
```
Comment (mathisloge) [inline ports/poco/0007-find-pcre2.patch:1]: this patch isn't needed anymore or?

Comment (MonicaLiu0311) [reply ports/poco/0007-find-pcre2.patch:1]: @FrankXie05 
> diff --git a/CMakeLists.txt b/CMakeLists.txt

I found the modification upstream, please check this patch.

Comment (mathisloge) [reply ports/poco/0007-find-pcre2.patch:1]: sorry. missed the removal in the `Foundation/CMakeLists.txt`

Comment (FrankXie05) [reply ports/poco/0007-find-pcre2.patch:1]: > @FrankXie05
> 
> > diff --git a/CMakeLists.txt b/CMakeLists.txt
> 
> I found the modification upstream, please check this patch.

I will update the patch next time a new version is released upstream.

**Comment by JavierMatosD** [comment]
Thank you!

**Pull request merged**